Program/src/file_delivery_app.c

```diff
@@ -0,0 +1,127 @@
+/*------------------------------------------------------------------------------
+This file is protected under copyright. If you want to use it,
+please include this text, that is my only stipulation.  
+
+Author: Evan Giese
+------------------------------------------------------------------------------*/
+#include "mib.h"
+#include "port.h"
+#include "file_delivery_app.h"
+#include "tasks.h"
+
+static void create_ssp_server_drivers(FTP *app) {
+
+    if (app->remote_entity.type_of_network == posix_connection_less) {
+        app->server_handle = ssp_thread_create(STACK_ALLOCATION, ssp_connectionless_server_task, app);
+
+    } else if(app->remote_entity.type_of_network == posix_connection) {
+        app->server_handle = ssp_thread_create(STACK_ALLOCATION, ssp_connection_server_task, app);
+
+    } else if (app->remote_entity.type_of_network == csp_connectionless) {
+        app->server_handle = ssp_thread_create(STACK_ALLOCATION, ssp_csp_connectionless_server_task, app);
+
+    } else if (app->remote_entity.type_of_network == csp_connection) {
+        app->server_handle = ssp_thread_create(STACK_ALLOCATION, ssp_csp_connection_server_task, app);
+    }
```

Comment:
great idea. Switch statements are much cleaner. It didn't occur to me, before I had multiple conditions for these, but now there is only one. A switch statement would be nicer.

Reply:
Just make sure the `default:` does something useful if you think it can be hit. E.g., logs a message like "unexpected operational parameter XYZ value"